src/datatypes/dtype.rs

```diff
@@ -11,6 +11,108 @@ use serde::{Deserialize, Serialize};
 
 // pub type TimeZone = String;
 
+#[allow(clippy::upper_case_acronyms)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Serialize, Deserialize, Hash)]
+pub enum ImageMode {
+    B = 1,
+    L = 2,
+    P = 3,
+    LA = 4,
+    RGB = 5,
+    RGBA = 6,
+    CMYK = 7,
+    YCbCr = 8,
+    LAB = 9,
+    HSV = 10,
+    I = 11,
+    F = 12,
```

Comment:
Not a fan of the generic `F` unless it's limited to  1 channel like in PIL. We have `RGB32F` and whatnot, what do you image is the usecase here?

Reply:
Basically just a single-channel float32 image, similar to PIL; but agreed, I think we should remove it and `I` as well.